File: src/main/java/coffee/amo/interactionkey/api/InteractableEntity.java
package coffee.amo.interactionkey.api;

import net.minecraft.nbt.CompoundTag;
import net.minecraft.network.protocol.Packet;
import net.minecraft.world.entity.Entity;
import net.minecraft.world.entity.EntityType;
import net.minecraft.world.entity.player.Player;
import net.minecraft.world.level.Level;

// Also does nothing unless you call it inside the interact event
public class InteractableEntity extends Entity implements Interactable {
    public Entity entity;

    public InteractableEntity(EntityType<?> pEntityType, Level pLevel) {
        super(pEntityType, pLevel);
    }

    @Override
    protected void defineSynchedData() {

    }

    @Override
    protected void readAdditionalSaveData(CompoundTag pCompound) {

    }

    @Override
    protected void addAdditionalSaveData(CompoundTag pCompound) {

    }

    @Override
    public String getInteractableName() {
        return null;
    }

    @Override
    public Packet<?> getAddEntityPacket() {
        return null;
    }

    @Override
    public String getDescription() {
        return null;
    }

    @Override
    public int getPriority() {
        return 0;
    }

    @Override
    public void interact(Player player) {

    }
}

File: src/main/kotlin/coffee/amo/interactionkey/api/InteractableEntity.kt
package coffee.amo.interactionkey.api

import net.minecraft.nbt.CompoundTag
import net.minecraft.network.protocol.Packet
import net.minecraft.world.entity.Entity
import net.minecraft.world.entity.EntityType
import net.minecraft.world.entity.player.Player
import net.minecraft.world.level.Level

// Also does nothing unless you call it inside the interact event
class InteractableEntity(pEntityType: EntityType<*>, pLevel: Level) : Entity(pEntityType, pLevel), Interactable {
    var entity: Entity? = null
    override fun defineSynchedData() {}
    override fun readAdditionalSaveData(pCompound: CompoundTag) {}
    override fun addAdditionalSaveData(pCompound: CompoundTag) {}
    override val interactableName: String?
        get() = null

    override fun getAddEntityPacket(): Packet<*>? {
        return null
    }

    override val description: String?
        get() = null
    override val priority: Int
        get() = 0

    override fun interact(player: Player?) {}
}